bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/SuspendMembershipFlow.kt

```diff
@@ -31,15 +31,21 @@ class SuspendMembershipFlow(val membership : StateAndRef<MembershipState<Any>>)
         // build suspension transaction
         val builder = TransactionBuilder(notary)
                 .addInputState(membership)
-                .addOutputState(membership.state.data.copy(status = MembershipStatus.SUSPENDED, modified = serviceHub.clock.instant()), configuration.membershipContractName())
+                .addOutputState(membership.state.data.copy(status = MembershipStatus.SUSPENDED, modified = serviceHub.clock.instant()))
                 .addCommand(MembershipContract.Commands.Suspend(), ourIdentity.owningKey)
         builder.verify(serviceHub)
         val selfSignedTx = serviceHub.signInitialTransaction(builder)
 
-        val finalisedTx = subFlow(FinalityFlow(selfSignedTx))
+        val memberSession = initiateFlow(membership.state.data.member)
+
+        val finalisedTx = if (memberSession.getCounterpartyFlowInfo().flowVersion == 1) {
+            subFlow(FinalityFlow(selfSignedTx))
+        } else {
+            subFlow(FinalityFlow(selfSignedTx, memberSession))
+        }
 
         val dbService = serviceHub.cordaService(DatabaseService::class.java)
-        val suspendedMembership = dbService.getMembership(membership.state.data.member, ourIdentity, configuration.membershipContractName())!!
+        val suspendedMembership = dbService.getMembership(membership.state.data.member, ourIdentity)!!
```

Comment:
Could `.let` and rethrow more meaningful exception

Reply:
Good spot, fixed